protocols/v2/roles-logic-sv2/src/utils.rs

```diff
@@ -739,6 +741,48 @@ fn tx_hash_list_hash_builder(txid_list: Vec<bitcoin::Txid>) -> U256<'static> {
     hash.to_vec().try_into().unwrap()
 }
 
+pub fn submit_solution_to_block(
```

Comment:
can you use From<SubmitSolution> for Block here?

Reply:
also having one or two test for it wouldn't be bad

Reply:
I don't think so, `From` trait has a non-compatible signature. Perhaps I don't understand you

Reply:
done